api/apps/api/src/modules/scenarios/marxan-run/run.service.ts

```diff
@@ -42,13 +28,16 @@ export class RunService {
     @InjectRepository(Scenario)
     private readonly scenarios: Repository<Scenario>,
     private readonly assets: AssetsService,
+    @Inject(blmDefaultToken)
+    private readonly blmDefault: number,
   ) {
     queueEvents.on(`completed`, ({ jobId }) => this.handleFinished(jobId));
     queueEvents.on(`failed`, ({ jobId }) => this.handleFailed(jobId));
   }
 
-  async run(scenarioId: string): Promise<void> {
-    const assets = await this.assets.forScenario(scenarioId);
+  async run(scenarioId: string, blm?: number): Promise<void> {
+    blm ??= this.blmDefault;
+    const assets = await this.assets.forScenario(scenarioId, blm);
```

Comment:
I am still not sure about this default and place where we apply it. Right now, we decide about final BLM value in two places (run.service and scenario.service below).

```
//scenarios.service.ts
blm ?? scenario.boundaryLengthModifier ?? blmDefault
```

or maybe even force BLM to be non-nullable and insert default upon creating a scenario? (@hotzevzl ?)

Reply:
Initially, I've implemented it like this, but I decided to keep defaulting to the `run.service` as it was easier to test, and didn't want to make another connection between scenario service and marxan params